src/java/org/apache/cassandra/index/sai/disk/StorageAttachedIndexWriter.java

```diff
@@ -164,6 +164,27 @@ public void staticRow(Row staticRow)
         }
     }
 
+    @Override
+    public void onSSTableWriterSwitched()
+    {
+        if (aborted) return;
+
+        try
+        {
+            for (PerColumnIndexWriter w : perIndexWriters)
+            {
+                w.onSSTableWriterSwitched(stopwatch);
+            }
+        }
+        catch (Throwable t)
+        {
+            logger.error(indexDescriptor.logMessage("Failed to flush segment on sstable writer switched"), t);
+            abort(t, true);
+            // fail compaction task or index build task if SAI failed
+            throw Throwables.unchecked(t);
```

Comment:
This is unnecessary. When `abort()` is called with `fromIndex == true`, as we do here, the exception will be thrown at the end of `abort()`.

Reply:
done